apps/protocol-frontend/src/contexts/UserContext.tsx

```diff
@@ -587,104 +613,134 @@ export const UserContextProvider: React.FC<UserContextProps> = ({
   return (
     <UserContext.Provider
       value={{
-        userAddress,
-        setUserAddress,
-        userData,
-        setUserData,
-        userDataByAddress,
-        setUserDataByAddress,
-        userContributions,
-        setUserContributions,
-        daoContributions,
-        setDaoContributions,
-        userAttestations,
-        setUserAttestations,
-        userActivityTypes,
-        setUserActivityTypes,
         allDaos,
-        getAllDaos,
+        authenticateAddress,
+        createAttestation,
+        createContribution,
         createUser,
         createWaitlistUser,
-        createContribution,
-        createAttestation,
-        mintContribution,
+        daoContributions,
+        getAllDaos,
+        isAuthenticated,
+        isAuthenticating,
         mintAttestation,
+        mintContribution,
+        setDaoContributions,
+        setUserActivityTypes,
+        setUserAddress,
+        setUserAttestations,
+        // setUserContributions,
+        setUserData,
+        setUserDataByAddress,
         updateContribution,
-        updateProfile,
         updateLinearEmail,
-        isAuthenticated,
-        isAuthenticating,
-        authenticateAddress,
+        updateProfile,
+        userActivityTypes,
+        userAddress,
+        userAttestations,
+        userContributions,
+        userData,
+        userDataByAddress,
       }}
     >
       {children}
     </UserContext.Provider>
   );
 };
+type UserContextType = {
+  allDaos: any;
+  authenticateAddress: any;
+  createAttestation: any;
+  createContribution: any;
+  createUser: any;
+  createWaitlistUser: any;
+  currentChain: any;
+  daoContributions: ContributionItem[];
+  isAuthenticated: boolean;
+  isAuthenticating: boolean;
+  mintAttestation: any;
+  mintContribution: any;
+  setAllDaos: any;
+  setCurrentChain: any;
+  setDaoContributions: any;
+  setUserActivityTypes: any;
+  setUserAddress: any;
+  setUserAttestations: any;
+  setUserData: any;
+  setUserDataByAddress: any;
+  updateContribution: any;
+  updateLinearEmail: any;
+  updateProfile: any;
+  userActivityTypes: any;
+  userAddress: any;
+  userAttestations: any;
+  userContributions: ContributionItem[];
+  userData: UserData;
+  userDataByAddress: any;
+};
```

Comment:
This is a good start, but we should narrow if possible

Reply:
This is strenuous task, I think, is going to take a lot of time. 
I am suggesting to gradually update `UserContextType`, while adding and supporting more types

Reply:
Also, It seems that User Context provider can't show errors for missing field in `UserContextType`; it continues to work. Not much help.

Reply:
I think a lot of these types already exist in our protocol-types. Let's take a look at this in a pairing session. If it is a lot of work we can make a new ticket to cover this